static/app/utils/profiling/hooks/useTransaction.tsx

```diff
@@ -0,0 +1,58 @@
+import {useEffect, useState} from 'react';
+
+import {Client} from 'sentry/api';
+import {Event, RequestState} from 'sentry/types';
+import useApi from 'sentry/utils/useApi';
+
+function fetchSentryEvent<T extends Event>(
+  api: Client,
+  organizationSlug: string,
+  projectSlug: string,
+  eventId: string
+): Promise<T> {
+  return api.requestPromise(
+    `/projects/${organizationSlug}/${projectSlug}/events/${eventId}/`
+  );
+}
+
+export function useSentryEvent<T extends Event>(
+  organizationSlug: string,
+  projectSlug: string,
+  eventId: string | null
+): RequestState<T> {
+  const api = useApi();
+  const [requestState, setRequestState] = useState<RequestState<T>>({
+    type: 'initial',
+  });
+
+  useEffect(() => {
+    if (eventId === null) {
+      return undefined;
+    }
+
+    let unmounted = false;
```

Comment:
you don't need this, `useApi` will handle canceling requests for you when it's unmounted. So you'll never reach the `then`

Reply:
wouldn't there still be a chance of hitting the `catch` block though? if thats the case we'd still want to avoid any `setState` calls

Reply:
When would it hit the catch block if it cancels the request before unmount?

Reply:
> wouldn't there still be a chance of hitting the `catch` block though? if thats the case we'd still want to avoid any `setState` calls

I'm assuming it cancels both resolve/reject, but I'm never sure so I just add the unmount flag.

Reply:
To me it looks like we'll catch an `AbortError` and still let it propagate to the `errorHandler` 

https://github.com/getsentry/sentry/blob/188f3c6a8e5f372ce4a9d184916eea5713a79cc7/static/app/api.tsx#L467-L494

if this is the case, i'd expect to see the `catch` in our promise to actually get this error, which would in turn leave us to potentially calling on `setState` depending on what we do with that error.